source/System.cpp

```diff
@@ -273,19 +252,21 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
 		}
 		else if(key == "belt")
 		{
-			Belt belt(child.Value(valueIndex),
-				child.Size() >= valueIndex + 2 ? max<int>(1, child.Value(valueIndex + 1)) : 1);
+			double radius = child.Value(valueIndex);
 			if(remove)
 			{
 				for(auto it = belts.begin(); it != belts.end(); ++it)
-					if(it->Radius() == belt.Radius())
+					if(*it == radius)
 					{
 						belts.eraseAt(it);
 						break;
 					}
```

Comment:
This section could be replaced with the following
```
erase(belts, radius)
```
but that would change the behavior from "erase the first match" to "erase all matches." But this is actually something that was discussed on #4704, where the game's current behavior for erasing variants from fleets is "erase the first match," and #4704 changes (and later #6592 did change) that behavior to "erase all matches."

I think that all such `remove *` event changes should erase all matches instead of erasing only the first match, since it's likely a content error to have duplicates of the same item.